### PR TITLE
Log QueryBuilder queries

### LINKED ISSUE #1676: log_query config setting is not respected by query builder
When looking trough some requests I noticed I was missing some queries in the query log.
It turns our that all queries that are handled by the query builder (the way we want it) are not logged.

CC: @nickvergessen @MorrisJobke

### PR DESCRIPTION
**Pull request by nickvergessen**
@LukasReschke @rullzer 

Fix #1676

### CODE REVIEW

File: lib/private/DB/QueryBuilder/QueryBuilder.php [hunk 1]
```diff
@@ -139,6 +139,29 @@ public function getState() {
 	 * @return \Doctrine\DBAL\Driver\Statement|int
 	 */
 	public function execute() {
+		if (\OC::$server->getConfig()->getSystemValue('log_query', false)) {
```
Comment (LukasReschke) [inline lib/private/DB/QueryBuilder/QueryBuilder.php]: Can't we inject that in \OC\DB\Connection::getQueryBuilder ?

Comment (nickvergessen) [inline lib/private/DB/QueryBuilder/QueryBuilder.php]: The problem is, that config needs database, so we need to be careful, but let me try.

**Comment by LukasReschke** [comment]
Tests failing now 🚀

**Comment by nickvergessen** [comment]
> Tests failing now

Fixed now, seems like data providers are called before `setUp()`

**Comment by rullzer** [comment]
LGTM

File: lib/private/DB/QueryBuilder/QueryBuilder.php [hunk 2]
```diff
@@ -139,6 +151,29 @@ public function getState() {
 	 * @return \Doctrine\DBAL\Driver\Statement|int
 	 */
 	public function execute() {
+		if ($this->systemConfig->getValue('log_query', false)) {
```
Comment (LukasReschke) [inline lib/private/DB/QueryBuilder/QueryBuilder.php]: Missing tests here. Will add…

File: lib/private/Security/CSRF/TokenStorage/SessionStorage.php [hunk 3]
```diff
@@ -41,6 +41,13 @@ public function __construct(ISession $session) {
 	}
 
 	/**
+	 * @param ISession $session
+	 */
+	public function setSession(ISession $session) {
```
Comment (LukasReschke) [inline lib/private/Security/CSRF/TokenStorage/SessionStorage.php]: Missing tests

**Review by LukasReschke — CHANGES_REQUESTED** [review]

**Review by LukasReschke — APPROVED** [review]

**Comment by LukasReschke** [comment]
Added the missing tests.

**Comment by rullzer** [comment]
YAY tests. :+1: stands

**Pull request merged**